File: 1018-binary-prefix-divisible-by-5/1018-binary-prefix-divisible-by-5.cpp
#pragma GCC optimize("O3", "unroll-loops");
static const int __ = []() {
    ios::sync_with_stdio(0);
    cin.tie(0);
    return 0;
}();
class Solution {
   public:
    vector<bool> prefixesDivBy5(vector<int>& nums) {
        int n = nums.size();
        int sum = 0;
        vector<bool> ans;
        for (int num : nums) {
            sum += num;
            sum %= 5;
            ans.emplace_back(sum ? false : true);
            sum <<= 1;
        }

        return ans;
    }
};

File: 1018-binary-prefix-divisible-by-5/1018-binary-prefix-divisible-by-5.go
func prefixesDivBy5(nums []int) []bool {
	n := len(nums)
	ans := make([]bool, 0, n)
	sum := 0

	for _, num := range nums {
		sum += num
		sum %= 5
		if sum == 0 {
			ans = append(ans, true)
		} else {
			ans = append(ans, false)
		}
		sum <<= 1
	}

	return ans
}
